unstructured/partition/pptx.py

```diff
@@ -156,13 +156,18 @@ def _iter_presentation_elements(self) -> Iterator[Element]:
             yield from self._increment_page_number()
             yield from self._iter_maybe_slide_notes(slide)
 
-            for shape in self._order_shapes(slide):
+            shapes, title_shape = self._order_shapes(slide)
+
+            for shape in shapes:
                 if shape.has_table:
                     assert isinstance(shape, GraphicFrame)
                     yield from self._iter_table_element(shape)
                 elif shape.has_text_frame:
                     assert isinstance(shape, Shape)
-                    yield from self._iter_paragraph_elements(shape)
+                    yield from self._iter_paragraph_elements(
+                        shape,
+                        is_title_shape=bool(shape == title_shape),
+                    )
```

Comment:
I'm inclined to think we would want to unconditionally process the title shape first, when it exists, because it provides hierarchical context for all the other shapes on the slide. Isn't that right? Or am I misunderstanding hierarchy or do we want to except shapes that appear above the title somehow?

If we want to process it separately, then `._order_shapes()` would still get this new signature, possibly `-> (title_shape: Shape | None, body_shapes: List[Shape])` giving title-shape pride of place, and then process title shape outside the loop when present.

Also in that case, I'd be inclined to make a new `._iter_title_shape_elements()` rather than add a flag argument to `._iter_paragraph_elements()`. That's Uncle Bob's Clean Code Tip #12 Avoid Boolean Flag Arguments although he's not the only one and perhaps not even the first to mention that in the literature:

https://martinfowler.com/bliki/FlagArgument.html
https://understandlegacycode.com/blog/what-is-wrong-with-boolean-parameters/

By the way, with the benefit of a few nights sleep and looking back on this, I'm thinking `._iter_paragraph_elements()` should really be `._iter_shape_elements()`, named for what it's decomposing rather than what it decomposes the shape into.

Reply:
I agree. Boolean didn't feel quite right here, felt like I was handjamming to fit your spec. I think an `_iter_title_shape_elements()`makes sense instead. I can also rename that function.